src/hooks/sync-sites-context.tsx

```diff
@@ -0,0 +1,212 @@
+import { useI18n } from '@wordpress/react-i18n';
+import React, { createContext, useContext, useState, useCallback, useEffect, useMemo } from 'react';
+import { getIpcApi } from '../lib/get-ipc-api';
+import { useAuth } from './use-auth';
+import { useImportExport } from './use-import-export';
+import { PullStateProgressInfo, useSyncStatesProgressInfo } from './use-sync-states-progress-info';
+
+export type SiteBackupState = {
+	backupId: string | null;
+	status: PullStateProgressInfo;
+	downloadUrl: string | null;
+	selectedSite: SiteDetails;
+};
+
+type SyncSitesContextType = {
+	pullStates: Record< number, SiteBackupState >;
+	pullSite: ( remoteSiteId: number, selectedSite: SiteDetails ) => Promise< void >;
+	isAnySitePulling: boolean;
+};
+
+const SyncSitesContext = createContext< SyncSitesContextType | undefined >( undefined );
+
+export function SyncSitesProvider( { children }: { children: React.ReactNode } ) {
+	const { pullStates, pullSite, isAnySitePulling } = useSyncPull();
+
+	return (
+		<SyncSitesContext.Provider value={ { pullStates, pullSite, isAnySitePulling } }>
+			{ children }
+		</SyncSitesContext.Provider>
+	);
+}
+
+function useSyncPull() {
```

Comment:
What do you think about moving this into a separate file and then passing it through the context? I am worried that once we add push operation and possible other things here, the file would become large and hard to manage.

Reply:
I believe it’s a great idea. I had concerns about the temptation to use the “smaller” hooks within other components, but I found an effective way to minimize the risk of misuse. I’ve moved the state into the context and passed it to useSyncPull, making it clear that the state is managed by a higher-level hook.

https://github.com/Automattic/studio/pull/612/commits/a22422f58a3579c7c9d939ba48635a91ce44a95d